mirrord/cli/src/connection.rs

```diff
@@ -35,29 +35,13 @@ where
     P: Progress,
     R: Reporter,
 {
-    let operator_required = match config.operator {
-        Some(true) => true,
-        Some(false) => return Ok(None),
-        None => false,
-    };
-
     let mut operator_subtask = progress.subtask("checking operator");
 
-    let mut find_operator_subtask = operator_subtask.subtask("detecting operator");
-
-    let api = if operator_required {
-        OperatorApi::new(config, analytics).await?.into()
-    } else {
-        OperatorApi::try_new(config, analytics).await?
-    };
-    let Some(mut api) = api else {
-        find_operator_subtask.success(Some("operator not found"));
+    let Some(api) = OperatorApi::try_new(config, analytics, &operator_subtask).await? else {
         operator_subtask.success(Some("proceeding without operator"));
```

Comment:
Feels like this may be a bit much first it's "checking operator" and inside of it it's "detecting operator" I would keep only one of them